topics/bloomfilters.md

```diff
@@ -0,0 +1,253 @@
+
+There other bloom filter commands are O(1) time complexity: BF.CARD, BF.INFO, BF.RESERVE, and BF.INSERT (when no items are provided).
+
+## Monitoring 
+
+To check the overall bloom filter metrics not just for specific filters, you can use the following `info bf` or `info modules`. 
+
+Example of `info bf` calls in different scenarios:
+
+```
+127.0.0.1:6379> info bf
+# bf_bloom_core_metrics
+bf_bloom_total_memory_bytes:0
+bf_bloom_num_objects:0
+bf_bloom_num_filters_across_objects:0
+bf_bloom_num_items_across_objects:0
+bf_bloom_capacity_across_objects:0
+
+# bf_bloom_defrag_metrics
+bf_bloom_defrag_hits:0
+bf_bloom_defrag_misses:0
+127.0.0.1:6379> bf.add key value
+(integer) 1
+127.0.0.1:6379> info bf
+# bf_bloom_core_metrics
+bf_bloom_total_memory_bytes:384
+bf_bloom_num_objects:1
+bf_bloom_num_filters_across_objects:1
+bf_bloom_num_items_across_objects:1
+bf_bloom_capacity_across_objects:100
+
+# bf_bloom_defrag_metrics
+bf_bloom_defrag_hits:0
+bf_bloom_defrag_misses:0
+```
+
+### Bloom filter core metrics
+
+* `bf_bloom_total_memory_bytes`: Current total number of bytes used by all bloom filters.
+
+* `bf_bloom_num_objects`: Current total number of bloom filters.
+
+* `bf_bloom_num_filters_across_objects`: Current total number of subfilters across all bloom filters.
+
+* `bf_bloom_num_items_across_objects`: Current total number of items across all bloom filters.
+
+* `bf_bloom_capacity_across_objects`: Current total capacity across all bloom filters.
+
+### Bloom filter defrag metrics
+
+* `bf_bloom_defrag_hits`: Total number of defrag hits that have occurred on bloom filters.
+
+* `bf_bloom_defrag_misses`: Total number of defrag misses that have occurred on bloom filters.
+
+## Limits
+
+There are two limits a bloom filter faces.
+
+1. Memory Usage Limit:
+
+    The memory usage limit per bloom filter by default is defined by the BF.BLOOM-MEMORY-USAGE-LIMIT module configuration which has a default value of 128 MB. If a command results in a creation / scale out causing the overall memory usage to exceed this limit, the command is rejected.
+
+2. Number of sub filters (in case of scalable bloom filters):
+
+    When a bloom filter scales out, a new sub filter is added. The limit on the number of sub filters depends on the false positive rate and tightening ratio. Each sub filter has a stricter false positive, and this is controlled by the tightening ratio. If a command attempting a scale out results in the sub filter reaching a false positive of 0, the command is rejected. 
+
+
+We have implemented VALIDATESCALETO as an optional arg of BF.INSERT to help determine whether the bloom filter can scale out to the reach the specified capacity without hitting either limits mentioned above, and will reject the command otherwise.
```

Comment:
```suggestion
We have implemented `VALIDATESCALETO` as an optional arg of `BF.INSERT` to help determine whether the bloom filter can scale out to the reach the specified capacity without hitting either limits mentioned above, and will reject the command otherwise.
```